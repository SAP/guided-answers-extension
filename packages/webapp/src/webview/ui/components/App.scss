html {
    height: 100%;
}
* {
    font-family: var(--vscode-font-family);
    -webkit-font-smoothing: antialiased;
}

h1 {
    font-weight: 300;
    color: var(--vscode-foreground);
    font-size: 24px;
}

h2 {
    font-size: 13px;
}

.guided-answer {
    &__header {
        overflow: hidden;
        position: fixed; /* Set the navbar to fixed position */
        top: 0; /* Position the navbar at the top of the page */
        width: 100%;
        z-index: 10;
        background-color: var(--vscode-editor-background);
        &__title {
            font-weight: 600;
            margin: 0px;
            color: var(--vscode-settings-headerForeground);
            line-height: 29px;
        }
        &__subtitle {
            color: var(--vscode-editorCodeLens-foreground, #999);
            font-size: 2em;
        }
        &__goback {
            height: 26px;
            margin-top: 17px;
        }
        &__logoAndTitle {
            display: flex;
            align-items: center;
            margin-top: 40px;
        }
        &__searchField {
            display: flex;
            align-items: center;
            flex-direction: column;
            margin-top: -28px;
            margin-bottom: 40px;
        }
    }
    &__container {
        margin-top: 111px;
        padding-bottom: 10px;
    }
    &__tree {
        width: 100%;
        height: auto;
        cursor: pointer;
        text-align: left;
        display: flex;
        border: none;
        &:hover {
            background-color: var(--vscode-list-hoverBackground);
        }

        &__ul {
            padding: 15px;
            &h3 {
                margin-bottom: 10px;
            }
        }
        &__title {
            font-size: 14px;
            color: var(--vscode-settings-headerForeground);
            margin-bottom: 10px;
            margin-top: 0px;
        }
        &__desc {
            color: var(--vscode-foreground);
            height: auto;
            font-size: 13px;
        }
    }
}

.tree-search-field {
    width: 33%;
    min-width: 250px;
}

.tree-search-icon {
    margin-top: 5px;
    text-align: center;
}

.tree-item {
    margin: 0;
}

#sap-logo {
    width: 59px;
}

.striped-list {
    padding-left: 0px;
    margin-top: 0px;
}

ul.striped-list > li:nth-of-type(odd) {
    background-color: var(--vscode-breadcrumbPicker-background);
    padding-left: 0px;
}

li {
    background-color: var(--vscode-editor-background);
    list-style-type: none;
}
