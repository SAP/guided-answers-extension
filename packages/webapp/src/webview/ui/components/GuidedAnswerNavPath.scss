* {
    box-sizing: border-box;
    outline: none;
}

h1 span {
    font-weight: 600;
}

.container {
    width: 100%;
    margin-top: 107px;
    position: relative;
    overflow: hidden;
    padding-left: 20px;
    padding-right: 20px;
}

.vertical-line {
    display: block;
    left: 9px;
    width: 1px;
    position: absolute;
    top: 47px;
    margin-left: -1px;
    height: 55px;
    background: lightgray;
    z-index: 1;
}

.timeline-block {
    width: 100%;
    margin: 0px 0px 0px 10px;
    display: flex;
    justify-content: space-between;
    clear: both;
    position: relative;
    float: none;
    height: 80px;
}

.marker {
    position: absolute;
    top: 20px;
    font-size: 20px;
}

.timeline-content {
    width: 100%;
    margin: 10px 10px 10px 35px;
    border: 1px solid lightgray;
    color: #666;
}

.timeline-content-arrow {
    position: absolute;
    top: 16px;
    left: -7px;
    position: relative;
    width: 0.75rem;
    height: 0.75rem;
    background-color: var(--vscode-editor-background);
    opacity: 1;
    -moz-opacity: 1;
    -webkit-opacity: 1;
    border-bottom: 0.0625rem solid #d1d1d1;
    border-left: 0.0625rem solid #d1d1d1;
    transform: translateX(0px) translateY(0px) rotate(45deg);
    -moz-transform: translateX(0px) translateY(0px) rotate(45deg);
    z-index: 1;
}

.timeline__path {
    display: inline-block;
    color: var(--vscode-textLink-foreground);
    cursor: pointer;
    padding: 1px 7px 2px 18px;
}
