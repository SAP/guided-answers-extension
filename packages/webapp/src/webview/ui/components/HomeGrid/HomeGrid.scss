.guided-answer__home-grid {
    display: flex;
    justify-content: center;
    gap: 5%;
    margin: 0 4%;

    &__column {
        flex: 0 1 570px;

        >div {
            margin-bottom: 30px;
        }
    }

    &__section {
        &__title {
            display: flex;
            align-items: center;
            margin: 20px 0;

            >i {
                height: 16px;
            }

            >span {
                margin: 0 5px;
            }
        }

        &__list {
            display: flex;
            flex-flow: column nowrap;
            gap: 5px;
            margin: 0;
            padding: 0;
            list-style: none;
        }
    }
}

@media only screen and (max-width: 740px) {
    .guided-answer__home-grid {
        display: block;
    }
}
