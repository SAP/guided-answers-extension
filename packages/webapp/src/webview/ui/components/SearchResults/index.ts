export * from './SearchResults';
export * from './SearchResultsTree';
