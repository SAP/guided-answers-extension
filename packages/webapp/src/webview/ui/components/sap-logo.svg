<svg width="53" height="26" viewBox="0 0 53 26" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M0 0V26H26.7843L53 0H0ZM32.2961 15.9976H30.6111V20.9663H22.884L22.1174 18.5889C21.2997 18.8549 20.3749 19.0038 19.3975 19.0038C18.42 19.0038 17.4457 18.8454 16.612 18.5667L15.8166 20.9663H11.0795L11.6017 19.5788C11.5394 19.6342 11.4772 19.6897 11.4117 19.7419C10.2298 20.6844 8.75241 21.2039 6.92047 21.2356H6.64896C4.5471 21.2372 2.48677 20.6622 0.744274 19.6802L2.3558 16.506C4.10149 17.5292 5.19555 17.7573 6.67611 17.7288C7.44274 17.7145 7.99536 17.5751 8.35791 17.2045C8.57513 16.978 8.69332 16.6992 8.69811 16.3983C8.71248 15.5731 7.53059 15.1866 6.08356 14.7431C4.90966 14.3804 3.57763 13.8894 2.53788 13.1323C1.30328 12.2326 0.715526 11.108 0.744274 9.51934C0.76344 8.36467 1.19467 7.32403 1.99325 6.50198C3.14001 5.32671 4.99111 4.61237 7.21116 4.61237H7.22393C9.12774 4.61553 11.45 5.14931 13.1334 5.99988L11.3174 9.13128C9.47752 8.24904 8.55117 8.16668 7.50184 8.11124C5.94142 8.02729 5.15083 8.58166 5.14124 9.1788C5.12847 9.90582 6.61542 10.5663 8.02411 11.013C10.1531 11.6846 12.8587 12.5826 13.2708 15.1739L17.1598 4.88638H21.6782L26.2365 17.1871V4.88638H31.4528C36.4391 4.88638 38.7821 6.65087 38.7821 10.4919C38.7821 13.9416 36.3704 15.9976 32.2961 15.9976V15.9976Z" fill= var(--vscode-settings-headerForeground)/>
<path d="M31.7593 8.27124H30.6094V12.4417H31.7593C33.2926 12.4417 34.5128 11.9364 34.5128 10.3319C34.5128 8.72741 33.2926 8.27124 31.7593 8.27124Z" fill= var(--vscode-settings-headerForeground)/>
<path d="M19.3783 9.9502L17.7236 15.1233C18.2156 15.3086 18.7873 15.4163 19.3959 15.4163C20.0044 15.4163 20.5458 15.3133 21.0329 15.1391L19.4134 9.95178H19.3799L19.3783 9.9502Z" fill= var(--vscode-settings-headerForeground)/>
</svg>
