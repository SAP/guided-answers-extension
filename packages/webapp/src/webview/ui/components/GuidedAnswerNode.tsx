import { default as parse, DOMNode, Element, Text } from 'html-react-parser';
import React, { ReactElement } from 'react';
import { GuidedAnswerNode, HTML_ENHANCEMENT_DATA_ATTR_MARKER } from '@sap/guided-answers-extension-types';
import { useSelector } from 'react-redux';
import { actions } from '../../state';
import { AppState } from '../../types';
import './GuidedAnswerNode.scss';
import { GuidedAnswerNavPath } from './GuidedAnswerNavPath';

/**
 * Replacer function for html-react-parser's replace function. If an element was marked, replace it with  link <a>
 * The command JSON is in the data-* attribute.
 *
 * @param domNode - current DOM node from html-react-parser
 * @returns - undefined if nothing to replace; the new node (<a>) in case of replacement
 */
function replace(domNode: DOMNode): JSX.Element | undefined {
    if (domNode.type === 'tag') {
        const domElement: Element = domNode as Element;
        const dataCommandString = domElement?.attribs?.[HTML_ENHANCEMENT_DATA_ATTR_MARKER];
        if (dataCommandString) {
            try {
                const command = JSON.parse(decodeURIComponent(dataCommandString));
                const textContent = domElement?.firstChild?.type === 'text' ? (domElement.firstChild as Text).data : '';
                if (command) {
                    return (
                        <a
                            href="javascript:void(0)"
                            onClick={(): void => {
                                actions.executeCommand(command);
                            }}>
                            {textContent}
                        </a>
                    );
                }
            } catch (error) {
                console.error(error);
            }
        }
    }
}

/**
 * Process the enhancements in an HTML string, which includes parsing the HTML and replacing the respective nodes.
 *
 * @param htmlString - HTML string
 * @returns - react elements including enhancements
 */
function enhanceBodyHtml(htmlString: string): ReactElement | undefined {
    let result;
    try {
        result = parse(htmlString, { replace });
    } catch (error) {
        console.error(error);
    }
    return result as ReactElement;
}

/**
 * Check if an HTML string contains an enhancement.
 *
 * @param htmlString - HTML string
 * @returns true: has enhancementsl; false: no enhancements
 */
function hasEnhancements(htmlString: string): boolean {
    return typeof htmlString === 'string' && htmlString.includes(HTML_ENHANCEMENT_DATA_ATTR_MARKER);
}

/**
 * Return the navigation section.
 *
 * @returns - react elements for navigation
 */
function getNavigationSection(): ReactElement {
    return (
        <div id="left" className="column">
            <GuidedAnswerNavPath />
        </div>
    );
}

/**
 * Return the content for a Guided Answers node, which consists of the middle part, the actual Guided Answers text
 * and, if present, the right part with additional commands.
 *
 * @param activeNode - the active Guided Answers node
 * @returns - react element for content
 */
function getContent(activeNode: GuidedAnswerNode): ReactElement {
    const enhancedBody = hasEnhancements(activeNode.BODY) ? enhanceBodyHtml(activeNode.BODY) : null;
    const middle = (
        <div id="middle" className="column">
            <h1>
                {activeNode.TITLE} <span className="light-text">#{activeNode.NODE_ID}</span>
            </h1>
            {enhancedBody ? enhancedBody : <div dangerouslySetInnerHTML={{ __html: activeNode.BODY }}></div>}
            <p className="guided-answer__node__question">{activeNode.QUESTION}</p>
            <div className="guided-answer__node">
                {activeNode.EDGES.map((edge, index) => (
                    <div
                        key={`edge_button${index}`}
                        className="guided-answer__node__edge"
                        onClick={(): void => {
                            actions.selectNode(edge.TARGET_NODE);
                        }}>
                        {edge.LABEL}
                    </div>
                ))}
            </div>
        </div>
    );
    const right = activeNode.COMMANDS ? (
        <div id="right" className="column">
            <div className="guided-answer__node__commands">
                {activeNode.COMMANDS
                    ? activeNode.COMMANDS.map((command, index) => (
                          <div className="guided-answer__node__command">
                              <div className="guided-answer__node__command__header">
                                  {/* <div className="guided-answer__node__command__header__icon">
                              {command.icon}
                          </div> */}
                                  <div className="guided-answer__node__command__header__label">{command.label}</div>
                              </div>

                              <div
                                  key={`command-${index}`}
                                  className="guided-answer__node__command__description"
                                  onClick={(): void => {
                                      actions.executeCommand(command);
                                  }}>
                                  {command.description}
                              </div>
                          </div>
                      ))
                    : ''}
            </div>
        </div>
    ) : null;

    return right ? (
        <div className="main-container">
            {middle}
            {right}
        </div>
    ) : (
        middle
    );
}

/**
 * Render the react elements to display a Guided Answers node.
 *
 * @returns - react element of Guided Answers node
 */
export function GuidedAnswerNode(): ReactElement {
    const nodes = useSelector<AppState, GuidedAnswerNode[]>((state) => state.activeGuidedAnswerNode);
    const activeNode = nodes[nodes.length - 1];
    return activeNode ? (
        <section className="guided-answer__node__body">
            {getNavigationSection()}
            {getContent(activeNode)}
        </section>
    ) : (
        <></>
    );
}
