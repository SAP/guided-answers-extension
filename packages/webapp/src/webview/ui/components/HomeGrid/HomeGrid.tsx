import type { ReactElement } from 'react';
import React from 'react';
import { useSelector } from 'react-redux';
import type { AppState } from '../../../types';
import { Bookmarks } from '../Bookmarks';
import { LastVisited } from '../LastVisited';
import { QuickFilters } from '../QuickFilters';
import './HomeGrid.scss';

/**
 * Home grid for Guided Answers Extension app.
 *
 * @returns - react elements for the home grid
 */
export function HomeGrid(): ReactElement {
    const appState = useSelector<AppState, AppState>((state) => state);

    const hasLastVisited = !!appState.lastVisitedGuides.length;
    const hasBookmarks = !!Object.keys(appState.bookmarks).length;
    const hasQuickFilters = !!appState.quickFilters.length;

    const isTwoColumnLayout = hasLastVisited && hasBookmarks && hasQuickFilters;

    return isTwoColumnLayout ? (
        <div className="guided-answer__home-grid">
            <div className="guided-answer__home-grid__column">
                <Bookmarks />
            </div>
            <div className="guided-answer__home-grid__column">
                <LastVisited />
                <QuickFilters />
            </div>
        </div>
    ) : (
        <div className="guided-answer__home-grid">
            <div className="guided-answer__home-grid__column">
                {hasLastVisited && <LastVisited />}
                {hasBookmarks && <Bookmarks />}
                {hasQuickFilters && <QuickFilters />}
            </div>
        </div>
    );
}
