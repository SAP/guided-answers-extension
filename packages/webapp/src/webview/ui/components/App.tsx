import type { ReactElement } from 'react';
import React from 'react';
import { useSelector } from 'react-redux';
import { VscSearch } from 'react-icons/vsc';
import { VSCodeButton, VSCodeTextField } from '@vscode/webview-ui-toolkit/react';
import { AppState } from '../../types';
import './App.scss';
import { actions } from '../../state';
import { GuidedAnswerNode } from './GuidedAnswerNode';

import Logo from './sap-logo.svg';

let timer: NodeJS.Timeout;

/**
 * Start element for Guided Answers Extension app.
 *
 * @returns - react elements for this app
 */
export function App(): ReactElement {
    const appState = useSelector<AppState, AppState>((state) => state);

    let content;
    if (appState.activeGuidedAnswerNode.length > 0) {
        content = <GuidedAnswerNode />;
    } else if (appState.guidedAnswerTrees) {
        content = (
            <>
                <VSCodeTextField
                    className="tree-search-field"
                    value={appState.query}
                    placeholder="Search Guided Answers"
                    onInput={(e: any) => {
                        console.log(`Value changed`, e);
                        const newValue = e?.target?.value;
                        if (newValue !== undefined) {
                            clearTimeout(timer);
                            actions.setQueryValue(newValue);
                            timer = setTimeout(() => {
                                actions.searchTree(newValue);
                            }, 400);
                        }
                    }}>
                    <span slot="end" className="tree-search-icon">
                        <VscSearch />
                    </span>
                </VSCodeTextField>
                {appState.guidedAnswerTrees.map((tree, index) => {
                    return (
                        <div key={`tree-item-${index}`} className="tree-item">
                            <div
                                className="guided-answer__tree"
                                onClick={(): void => {
                                    actions.setActiveTree(tree);
                                    actions.selectNode(tree.FIRST_NODE_ID);
                                }}>
                                <ul className="guided-answer__tree__ul">
                                    <h3 className="guided-answer__tree__title">{tree.TITLE}</h3>
                                    {tree.DESCRIPTION && (
                                        <span className="guided-answer__tree__desc">{tree.DESCRIPTION}</span>
                                    )}
                                </ul>
                            </div>
                        </div>
                    );
                })}
            </>
        );
    }
    const backButton =
        appState.activeGuidedAnswerNode.length > 0 ? (
            <VSCodeButton
                className="guided-answer__header__goback"
                onClick={(): void => {
                    actions.goToPreviousPage();
                }}>
                Go back
            </VSCodeButton>
        ) : (
            <></>
        );
    return (
        <div className="guided-answer">
            <div className="guided-answer__header">
                <div className="guided-answer__header__sub">
                    <span id="sap-logo">
                        <Logo />
                    </span>
                    <h1 className="guided-answer__header__title">Guided Answers</h1>
                    <span className="guided-answer__header__subtitle">
                        {appState.activeGuidedAnswer ? ': ' + appState.activeGuidedAnswer.TITLE : ''}
                    </span>
                </div>
                {backButton}
            </div>
            <main className="guided-answer__container">{content}</main>
        </div>
    );
}
