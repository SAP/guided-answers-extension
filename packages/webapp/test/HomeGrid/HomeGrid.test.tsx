import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { useSelector } from 'react-redux';
import { HomeGrid } from '../../src/webview/ui/components/HomeGrid';

jest.mock('react-redux', () => ({
    useSelector: jest.fn()
}));

describe('<HomeGrid />', () => {
    afterEach(cleanup);

    it('Should render only Bookmarks', () => {
        (useSelector as jest.Mock).mockImplementation((selector) =>
            selector({
                bookmarks: {
                    'tree1-node1': {
                        tree: {
                            TREE_ID: 'tree1',
                            TITLE: 'Bookmark 1 Title',
                            DESCRIPTION: 'Bookmark 1 Description',
                            PRODUCT: 'Product 1, Product 2',
                            COMPONENT: 'Component 1, Component 2'
                        },
                        nodePath: [{ NODE_ID: 'node1', TITLE: 'Node 1' }]
                    }
                },
                lastVisitedGuides: [],
                autoFilters: []
            })
        );
        const { container } = render(<HomeGrid />);
        expect(container).toMatchSnapshot();
    });

    it('Should render only LastVisited', () => {
        (useSelector as jest.Mock).mockImplementation((selector) =>
            selector({
                bookmarks: {},
                lastVisitedGuides: [
                    {
                        tree: {
                            TREE_ID: '1',
                            TITLE: 'Bookmark 1 Title',
                            DESCRIPTION: 'Bookmark 1 Description',
                            PRODUCT: 'Product 1, Product 2',
                            COMPONENT: 'Component 1, Component 2'
                        },
                        nodePath: [{ NODE_ID: '1', TITLE: 'Node 1' }],
                        createdAt: 'time'
                    }
                ],
                autoFilters: []
            })
        );
        const { container } = render(<HomeGrid />);
        expect(container).toMatchSnapshot();
    });

    it('Should render only Filters', () => {
        (useSelector as jest.Mock).mockImplementation((selector) =>
            selector({
                bookmarks: {},
                lastVisitedGuides: [],
                autoFilters: [
                    {
                        product: ['product 1'],
                        component: ['component 1']
                    }
                ]
            })
        );
        const { container } = render(<HomeGrid />);
        expect(container).toMatchSnapshot();
    });

    it('Should render all in 2 columns', () => {
        (useSelector as jest.Mock).mockImplementation((selector) =>
            selector({
                bookmarks: {
                    'tree1-node1': {
                        tree: {
                            TREE_ID: 'tree1',
                            TITLE: 'Bookmark 1 Title',
                            DESCRIPTION: 'Bookmark 1 Description',
                            PRODUCT: 'Product 1, Product 2',
                            COMPONENT: 'Component 1, Component 2'
                        },
                        nodePath: [{ NODE_ID: 'node1', TITLE: 'Node 1' }]
                    }
                },
                lastVisitedGuides: [
                    {
                        tree: {
                            TREE_ID: '1',
                            TITLE: 'Bookmark 1 Title',
                            DESCRIPTION: 'Bookmark 1 Description',
                            PRODUCT: 'Product 1, Product 2',
                            COMPONENT: 'Component 1, Component 2'
                        },
                        nodePath: [{ NODE_ID: '1', TITLE: 'Node 1' }],
                        createdAt: 'time'
                    }
                ],
                autoFilters: [
                    {
                        product: ['product 1'],
                        component: ['component 1']
                    }
                ]
            })
        );
        const { container } = render(<HomeGrid />);
        expect(container).toMatchSnapshot();
    });

    it('Should render only Bookmarks, no beta features', () => {
        (useSelector as jest.Mock).mockImplementation((selector) =>
            selector({
                bookmarks: {
                    'tree1-node1': {
                        tree: {
                            TREE_ID: 'tree1',
                            TITLE: 'Bookmark 1 Title',
                            DESCRIPTION: 'Bookmark 1 Description',
                            PRODUCT: 'Product 1, Product 2',
                            COMPONENT: 'Component 1, Component 2'
                        },
                        nodePath: [{ NODE_ID: 'node1', TITLE: 'Node 1' }]
                    }
                },
                lastVisitedGuides: [],
                autoFilters: []
            })
        );
        const { container } = render(<HomeGrid />);
        expect(container).toMatchSnapshot();
    });

    it('Should render only LastVisited, no beta features', () => {
        (useSelector as jest.Mock).mockImplementation((selector) =>
            selector({
                bookmarks: {},
                lastVisitedGuides: [
                    {
                        tree: {
                            TREE_ID: '1',
                            TITLE: 'Bookmark 1 Title',
                            DESCRIPTION: 'Bookmark 1 Description',
                            PRODUCT: 'Product 1, Product 2',
                            COMPONENT: 'Component 1, Component 2'
                        },
                        nodePath: [{ NODE_ID: '1', TITLE: 'Node 1' }],
                        createdAt: 'time'
                    }
                ],
                autoFilters: []
            })
        );
        const { container } = render(<HomeGrid />);
        expect(container).toMatchSnapshot();
    });
});
