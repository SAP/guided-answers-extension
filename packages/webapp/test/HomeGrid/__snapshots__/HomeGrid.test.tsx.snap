// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeGrid /> Should render all in 2 columns 1`] = `
<div>
  <div
    class="guided-answer__home-grid"
  >
    <div
      class="guided-answer__home-grid__column"
    >
      <div>
        <h3
          class="guided-answer__home-grid__section__title"
        >
          <i
            aria-hidden="true"
            class="ts-icon root-162"
            data-icon-name="StarActive"
          />
          <span>
            Bookmarks
          </span>
        </h3>
        <div
          class="ms-FocusZone css-166"
          data-focuszone-id="FocusZone6"
        >
          <ul
            class="guided-answer__home-grid__section__list"
          >
            <li
              class="guided-answer__bookmark"
            >
              <div
                class="guided-answer__bookmark__title"
              >
                <a
                  id="goto-bookmark-button"
                  tabindex="0"
                >
                  Bookmark 1 Title
                </a>
                <button
                  class="ms-Button ms-Button--icon root-167"
                  data-is-focusable="true"
                  id="toggle-bookmark-button"
                  tabindex="-1"
                  type="button"
                >
                  <span
                    class="ms-Button-flexContainer flexContainer-168"
                    data-automationid="splitbuttonprimary"
                  >
                    <i
                      aria-hidden="true"
                      class="ms-Icon root-162 ms-Button-icon icon-170"
                      data-icon-name="StarActive"
                    />
                  </span>
                </button>
              </div>
              <div
                class="bottom-section"
                id="bottom-section"
              >
                <div
                  class="component-and-product-container"
                  style="margin-top: 0px;"
                >
                  <div
                    class="guided-answer__tree__product"
                    id="product-container"
                  >
                    <span
                      class="bottom-title"
                    >
                      Product: 
                    </span>
                    Product 1
                  </div>
                  <div
                    class="guided-answer__tree__component"
                    id="component-container"
                  >
                    <span
                      class="bottom-title"
                    >
                      Component: 
                    </span>
                    Component 1
                  </div>
                </div>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="guided-answer__home-grid__column"
    >
      <div>
        <h3
          class="guided-answer__home-grid__section__title"
        >
          <i
            aria-hidden="true"
            class="ts-icon root-162"
            data-icon-name="History"
          />
          <span>
            Last visited
          </span>
        </h3>
        <div
          class="ms-FocusZone css-166"
          data-focuszone-id="FocusZone10"
        >
          <ul
            class="guided-answer__home-grid__section__list"
          >
            <li
              class="guided-answer__last-visited"
            >
              <a
                id="last-visited-link"
                tabindex="0"
              >
                Bookmark 1 Title
              </a>
              <div
                class="bottom-section"
                id="bottom-section"
              >
                <div
                  class="component-and-product-container"
                  style="margin-top: 0px;"
                >
                  <div
                    class="guided-answer__tree__product"
                    id="product-container"
                  >
                    <span
                      class="bottom-title"
                    >
                      Product: 
                    </span>
                    Product 1
                  </div>
                  <div
                    class="guided-answer__tree__component"
                    id="component-container"
                  >
                    <span
                      class="bottom-title"
                    >
                      Component: 
                    </span>
                    Component 1
                  </div>
                </div>
              </div>
            </li>
          </ul>
        </div>
      </div>
      <div>
        <h3
          class="guided-answer__home-grid__section__title"
        >
          <i
            aria-hidden="true"
            class="ts-icon root-162"
            data-icon-name="Filter"
          />
          <span>
            Quick Filters
          </span>
        </h3>
        <div
          class="ms-FocusZone css-166"
          data-focuszone-id="FocusZone11"
        >
          <ul
            class="guided-answer__home-grid__section__list"
          >
            <li
              class="guided-answer__quick-filter"
            >
              <button
                tabindex="0"
              >
                <div>
                  Product: 
                  <strong>
                    product 1
                  </strong>
                </div>
                <div>
                  Component: 
                  <strong>
                    component 1
                  </strong>
                </div>
              </button>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<HomeGrid /> Should render only Bookmarks 1`] = `
<div>
  <div
    class="guided-answer__home-grid"
  >
    <div
      class="guided-answer__home-grid__column"
    >
      <div>
        <h3
          class="guided-answer__home-grid__section__title"
        >
          <i
            aria-hidden="true"
            class="ts-icon root-162"
            data-icon-name="StarActive"
          />
          <span>
            Bookmarks
          </span>
        </h3>
        <div
          class="ms-FocusZone css-166"
          data-focuszone-id="FocusZone0"
        >
          <ul
            class="guided-answer__home-grid__section__list"
          >
            <li
              class="guided-answer__bookmark"
            >
              <div
                class="guided-answer__bookmark__title"
              >
                <a
                  id="goto-bookmark-button"
                  tabindex="0"
                >
                  Bookmark 1 Title
                </a>
                <button
                  class="ms-Button ms-Button--icon root-167"
                  data-is-focusable="true"
                  id="toggle-bookmark-button"
                  tabindex="-1"
                  type="button"
                >
                  <span
                    class="ms-Button-flexContainer flexContainer-168"
                    data-automationid="splitbuttonprimary"
                  >
                    <i
                      aria-hidden="true"
                      class="ms-Icon root-162 ms-Button-icon icon-170"
                      data-icon-name="StarActive"
                    />
                  </span>
                </button>
              </div>
              <div
                class="bottom-section"
                id="bottom-section"
              >
                <div
                  class="component-and-product-container"
                  style="margin-top: 0px;"
                >
                  <div
                    class="guided-answer__tree__product"
                    id="product-container"
                  >
                    <span
                      class="bottom-title"
                    >
                      Product: 
                    </span>
                    Product 1
                  </div>
                  <div
                    class="guided-answer__tree__component"
                    id="component-container"
                  >
                    <span
                      class="bottom-title"
                    >
                      Component: 
                    </span>
                    Component 1
                  </div>
                </div>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<HomeGrid /> Should render only Bookmarks, no beta features 1`] = `
<div>
  <div
    class="guided-answer__home-grid"
  >
    <div
      class="guided-answer__home-grid__column"
    >
      <div>
        <h3
          class="guided-answer__home-grid__section__title"
        >
          <i
            aria-hidden="true"
            class="ts-icon root-162"
            data-icon-name="StarActive"
          />
          <span>
            Bookmarks
          </span>
        </h3>
        <div
          class="ms-FocusZone css-166"
          data-focuszone-id="FocusZone12"
        >
          <ul
            class="guided-answer__home-grid__section__list"
          >
            <li
              class="guided-answer__bookmark"
            >
              <div
                class="guided-answer__bookmark__title"
              >
                <a
                  id="goto-bookmark-button"
                  tabindex="0"
                >
                  Bookmark 1 Title
                </a>
                <button
                  class="ms-Button ms-Button--icon root-167"
                  data-is-focusable="true"
                  id="toggle-bookmark-button"
                  tabindex="-1"
                  type="button"
                >
                  <span
                    class="ms-Button-flexContainer flexContainer-168"
                    data-automationid="splitbuttonprimary"
                  >
                    <i
                      aria-hidden="true"
                      class="ms-Icon root-162 ms-Button-icon icon-170"
                      data-icon-name="StarActive"
                    />
                  </span>
                </button>
              </div>
              <div
                class="bottom-section"
                id="bottom-section"
              >
                <div
                  class="component-and-product-container"
                  style="margin-top: 0px;"
                >
                  <div
                    class="guided-answer__tree__product"
                    id="product-container"
                  >
                    <span
                      class="bottom-title"
                    >
                      Product: 
                    </span>
                    Product 1
                  </div>
                  <div
                    class="guided-answer__tree__component"
                    id="component-container"
                  >
                    <span
                      class="bottom-title"
                    >
                      Component: 
                    </span>
                    Component 1
                  </div>
                </div>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<HomeGrid /> Should render only LastVisited 1`] = `
<div>
  <div
    class="guided-answer__home-grid"
  >
    <div
      class="guided-answer__home-grid__column"
    >
      <div>
        <h3
          class="guided-answer__home-grid__section__title"
        >
          <i
            aria-hidden="true"
            class="ts-icon root-162"
            data-icon-name="History"
          />
          <span>
            Last visited
          </span>
        </h3>
        <div
          class="ms-FocusZone css-166"
          data-focuszone-id="FocusZone4"
        >
          <ul
            class="guided-answer__home-grid__section__list"
          >
            <li
              class="guided-answer__last-visited"
            >
              <a
                id="last-visited-link"
                tabindex="0"
              >
                Bookmark 1 Title
              </a>
              <div
                class="bottom-section"
                id="bottom-section"
              >
                <div
                  class="component-and-product-container"
                  style="margin-top: 0px;"
                >
                  <div
                    class="guided-answer__tree__product"
                    id="product-container"
                  >
                    <span
                      class="bottom-title"
                    >
                      Product: 
                    </span>
                    Product 1
                  </div>
                  <div
                    class="guided-answer__tree__component"
                    id="component-container"
                  >
                    <span
                      class="bottom-title"
                    >
                      Component: 
                    </span>
                    Component 1
                  </div>
                </div>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<HomeGrid /> Should render only LastVisited, no beta features 1`] = `
<div>
  <div
    class="guided-answer__home-grid"
  >
    <div
      class="guided-answer__home-grid__column"
    >
      <div>
        <h3
          class="guided-answer__home-grid__section__title"
        >
          <i
            aria-hidden="true"
            class="ts-icon root-162"
            data-icon-name="History"
          />
          <span>
            Last visited
          </span>
        </h3>
        <div
          class="ms-FocusZone css-166"
          data-focuszone-id="FocusZone16"
        >
          <ul
            class="guided-answer__home-grid__section__list"
          >
            <li
              class="guided-answer__last-visited"
            >
              <a
                id="last-visited-link"
                tabindex="0"
              >
                Bookmark 1 Title
              </a>
              <div
                class="bottom-section"
                id="bottom-section"
              >
                <div
                  class="component-and-product-container"
                  style="margin-top: 0px;"
                >
                  <div
                    class="guided-answer__tree__product"
                    id="product-container"
                  >
                    <span
                      class="bottom-title"
                    >
                      Product: 
                    </span>
                    Product 1
                  </div>
                  <div
                    class="guided-answer__tree__component"
                    id="component-container"
                  >
                    <span
                      class="bottom-title"
                    >
                      Component: 
                    </span>
                    Component 1
                  </div>
                </div>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<HomeGrid /> Should render only QuickFilters 1`] = `
<div>
  <div
    class="guided-answer__home-grid"
  >
    <div
      class="guided-answer__home-grid__column"
    >
      <div>
        <h3
          class="guided-answer__home-grid__section__title"
        >
          <i
            aria-hidden="true"
            class="ts-icon root-162"
            data-icon-name="Filter"
          />
          <span>
            Quick Filters
          </span>
        </h3>
        <div
          class="ms-FocusZone css-166"
          data-focuszone-id="FocusZone5"
        >
          <ul
            class="guided-answer__home-grid__section__list"
          >
            <li
              class="guided-answer__quick-filter"
            >
              <button
                tabindex="0"
              >
                <div>
                  Product: 
                  <strong>
                    product 1
                  </strong>
                </div>
                <div>
                  Component: 
                  <strong>
                    component 1
                  </strong>
                </div>
              </button>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
